core/Archive.php

```diff
@@ -611,22 +619,23 @@ private function get($archiveNames, $archiveDataType, $idSubtable = null)
             return $result;
         }
 
-        $loadAllSubtables = $idSubtable == self::ID_SUBTABLE_LOAD_ALL_SUBTABLES;
-        $archiveData = ArchiveSelector::getArchiveData($archiveIds, $archiveNames, $archiveDataType, $loadAllSubtables);
+        $archiveData = ArchiveSelector::getArchiveData($archiveIds, $archiveNames, $archiveDataType, $idSubtable);
+
         foreach ($archiveData as $row) {
             // values are grouped by idsite (site ID), date1-date2 (date range), then name (field name)
-            $idSite = $row['idsite'];
+            $idSite    = $row['idsite'];
             $periodStr = $row['date1'] . "," . $row['date2'];
 
             if ($archiveDataType == 'numeric') {
-                $value = $this->formatNumericValue($row['value']);
+                $row['value'] = $this->formatNumericValue($row['value']);
             } else {
-                $value = $this->uncompress($row['value']);
```

Comment:
`uncompress()` was done here in `Archive` but `compress()` is done in `ArchiveWriter`. Therefore the `uncompress()` should as well done in `ArchiveSelector` (which is basically a `ArchiveReader`). `Archive` does in theory not know it is compressed and shouldn't care about that. This is in the end a db specifc / writer specific thing.